File: qt/gen_qjsonarray.go
package qt

import (
	"unsafe"
)

type QJsonArray struct {
	h          uintptr
	isSubclass bool
}

// NewQJsonArray constructs a new QJsonArray object.
func NewQJsonArray() *QJsonArray {

	ret := newQJsonArray(QJsonArray_new())
	ret.isSubclass = true
	return ret
}

// NewQJsonArray2 constructs a new QJsonArray object.
func NewQJsonArray2(other *QJsonArray) *QJsonArray {

	ret := newQJsonArray(QJsonArray_new2(other.cPointer()))
	ret.isSubclass = true
	return ret
}

func (this *QJsonArray) OperatorAssign(other *QJsonArray) {
	QJsonArray_OperatorAssign(this.h, other.cPointer())
}

func QJsonArray_FromStringList(list []string) *QJsonArray {
	list_CArray := (*[0xffff]struct_miqt_string)(malloc(size_t(int(unsafe.Sizeof(struct_miqt_string{})) * len(list))))
	defer free(unsafe.Pointer(list_CArray))
	for i := range list {
		list_i_ms := struct_miqt_string{}
		list_i_ms.data = CString(list[i])
		list_i_ms.len = size_t(len(list[i]))
		defer free(unsafe.Pointer(list_i_ms.data))
		list_CArray[i] = list_i_ms
	}
	list_ma := struct_miqt_array{len: size_t(len(list)), data: unsafe.Pointer(list_CArray)}
	_goptr := newQJsonArray(QJsonArray_FromStringList(list_ma))
	_goptr.GoGC() // Qt uses pass-by-value semantics for this type. Mimic with finalizer
	return _goptr
}

func (this *QJsonArray) Size() int64 {
	return (int64)(QJsonArray_Size(this.h))
}

func (this *QJsonArray) Count() int64 {
	return (int64)(QJsonArray_Count(this.h))
}

func (this *QJsonArray) IsEmpty() bool {
	return (bool)(QJsonArray_IsEmpty(this.h))
}

func (this *QJsonArray) At(i int64) *QJsonValue {
	_goptr := newQJsonValue(QJsonArray_At(this.h, (ptrdiff_t)(i)))
	_goptr.GoGC() // Qt uses pass-by-value semantics for this type. Mimic with finalizer
	return _goptr
}

func (this *QJsonArray) First() *QJsonValue {
	_goptr := newQJsonValue(QJsonArray_First(this.h))
	_goptr.GoGC() // Qt uses pass-by-value semantics for this type. Mimic with finalizer
	return _goptr
}

func (this *QJsonArray) Last() *QJsonValue {
	_goptr := newQJsonValue(QJsonArray_Last(this.h))
	_goptr.GoGC() // Qt uses pass-by-value semantics for this type. Mimic with finalizer
	return _goptr
}

func (this *QJsonArray) Prepend(value *QJsonValue) {
	QJsonArray_Prepend(this.h, value.cPointer())
}

func (this *QJsonArray) Append(value *QJsonValue) {
	QJsonArray_Append(this.h, value.cPointer())
}

func (this *QJsonArray) RemoveAt(i int64) {
	QJsonArray_RemoveAt(this.h, (ptrdiff_t)(i))
}

func (this *QJsonArray) TakeAt(i int64) *QJsonValue {
	_goptr := newQJsonValue(QJsonArray_TakeAt(this.h, (ptrdiff_t)(i)))
	_goptr.GoGC() // Qt uses pass-by-value semantics for this type. Mimic with finalizer
	return _goptr
}

func (this *QJsonArray) RemoveFirst() {
	QJsonArray_RemoveFirst(this.h)
}

func (this *QJsonArray) RemoveLast() {
	QJsonArray_RemoveLast(this.h)
}

func (this *QJsonArray) Insert(i int64, value *QJsonValue) {
	QJsonArray_Insert(this.h, (ptrdiff_t)(i), value.cPointer())
}

func (this *QJsonArray) Replace(i int64, value *QJsonValue) {
	QJsonArray_Replace(this.h, (ptrdiff_t)(i), value.cPointer())
}

func (this *QJsonArray) Contains(element *QJsonValue) bool {
	return (bool)(QJsonArray_Contains(this.h, element.cPointer()))
}

func (this *QJsonArray) OperatorSubscript(i int64) *QJsonValueRef {
	_goptr := newQJsonValueRef(QJsonArray_OperatorSubscript(this.h, (ptrdiff_t)(i)))
	_goptr.GoGC() // Qt uses pass-by-value semantics for this type. Mimic with finalizer
	return _goptr
}

func (this *QJsonArray) OperatorSubscriptWithQsizetype(i int64) *QJsonValue {
	_goptr := newQJsonValue(QJsonArray_OperatorSubscriptWithQsizetype(this.h, (ptrdiff_t)(i)))
	_goptr.GoGC() // Qt uses pass-by-value semantics for this type. Mimic with finalizer
	return _goptr
}

func (this *QJsonArray) Swap(other *QJsonArray) {
	QJsonArray_Swap(this.h, other.cPointer())
}

func (this *QJsonArray) Begin() iterator {
	xxxxxxxxx
}

func (this *QJsonArray) Begin2() const_iterator {
	xxxxxxxxx
}

func (this *QJsonArray) ConstBegin() const_iterator {
	xxxxxxxxx
}

func (this *QJsonArray) Cbegin() const_iterator {
	xxxxxxxxx
}

func (this *QJsonArray) End() iterator {
	xxxxxxxxx
}

func (this *QJsonArray) End2() const_iterator {
	xxxxxxxxx
}

func (this *QJsonArray) ConstEnd() const_iterator {
	xxxxxxxxx
}

func (this *QJsonArray) Cend() const_iterator {
	xxxxxxxxx
}

func (this *QJsonArray) Insert2(before iterator, value *QJsonValue) iterator {
	xxxxxxxxx
}

func (this *QJsonArray) Erase(it iterator) iterator {
	xxxxxxxxx
}

func (this *QJsonArray) OperatorPlus(v *QJsonValue) *QJsonArray {
	_goptr := newQJsonArray(QJsonArray_OperatorPlus(this.h, v.cPointer()))
	_goptr.GoGC() // Qt uses pass-by-value semantics for this type. Mimic with finalizer
	return _goptr
}

func (this *QJsonArray) OperatorPlusAssign(v *QJsonValue) *QJsonArray {
	return newQJsonArray(QJsonArray_OperatorPlusAssign(this.h, v.cPointer()))
}

func (this *QJsonArray) OperatorShiftLeft(v *QJsonValue) *QJsonArray {
	return newQJsonArray(QJsonArray_OperatorShiftLeft(this.h, v.cPointer()))
}

func (this *QJsonArray) PushBack(t *QJsonValue) {
	QJsonArray_PushBack(this.h, t.cPointer())
}

func (this *QJsonArray) PushFront(t *QJsonValue) {
	QJsonArray_PushFront(this.h, t.cPointer())
}

func (this *QJsonArray) PopFront() {
	QJsonArray_PopFront(this.h)
}

func (this *QJsonArray) PopBack() {
	QJsonArray_PopBack(this.h)
}

func (this *QJsonArray) Empty() bool {
	return (bool)(QJsonArray_Empty(this.h))
}

type QJsonArray__iterator struct {
	h          uintptr
	isSubclass bool
}

// NewQJsonArray__iterator constructs a new QJsonArray::iterator object.
func NewQJsonArray__iterator() *QJsonArray__iterator {

	ret := newQJsonArray__iterator(QJsonArray__iterator_new())
	ret.isSubclass = true
	return ret
}

// NewQJsonArray__iterator2 constructs a new QJsonArray::iterator object.
func NewQJsonArray__iterator2(array *QJsonArray, index int64) *QJsonArray__iterator {

	ret := newQJsonArray__iterator(QJsonArray__iterator_new2(array.cPointer(), (ptrdiff_t)(index)))
	ret.isSubclass = true
	return ret
}

// NewQJsonArray__iterator3 constructs a new QJsonArray::iterator object.
func NewQJsonArray__iterator3(other *iterator) *QJsonArray__iterator {

	ret := newQJsonArray__iterator(QJsonArray__iterator_new3(other))
	ret.isSubclass = true
	return ret
}

func (this *QJsonArray__iterator) OperatorAssign(other *iterator) {
	QJsonArray__iterator_OperatorAssign(this.h, other)
}

func (this *QJsonArray__iterator) OperatorMultiply() *QJsonValueRef {
	_goptr := newQJsonValueRef(QJsonArray__iterator_OperatorMultiply(this.h))
	_goptr.GoGC() // Qt uses pass-by-value semantics for this type. Mimic with finalizer
	return _goptr
}

func (this *QJsonArray__iterator) OperatorMinusGreater() *QJsonValueConstRef {
	return newQJsonValueConstRef(QJsonArray__iterator_OperatorMinusGreater(this.h))
}

func (this *QJsonArray__iterator) OperatorMinusGreater2() *QJsonValueRef {
	return newQJsonValueRef(QJsonArray__iterator_OperatorMinusGreater2(this.h))
}

func (this *QJsonArray__iterator) OperatorSubscript(j int64) *QJsonValueRef {
	_goptr := newQJsonValueRef(QJsonArray__iterator_OperatorSubscript(this.h, (ptrdiff_t)(j)))
	_goptr.GoGC() // Qt uses pass-by-value semantics for this type. Mimic with finalizer
	return _goptr
}

func (this *QJsonArray__iterator) OperatorPlusPlus() *iterator {
	xxxxxxxxx
}

func (this *QJsonArray__iterator) OperatorPlusPlusWithInt(param1 int) iterator {
	xxxxxxxxx
}

func (this *QJsonArray__iterator) OperatorMinusMinus() *iterator {
	xxxxxxxxx
}

func (this *QJsonArray__iterator) OperatorMinusMinusWithInt(param1 int) iterator {
	xxxxxxxxx
}

func (this *QJsonArray__iterator) OperatorPlusAssign(j int64) *iterator {
	xxxxxxxxx
}

func (this *QJsonArray__iterator) OperatorMinusAssign(j int64) *iterator {
	xxxxxxxxx
}

func (this *QJsonArray__iterator) OperatorPlus(j int64) iterator {
	xxxxxxxxx
}

func (this *QJsonArray__iterator) OperatorMinus(j int64) iterator {
	xxxxxxxxx
}

func (this *QJsonArray__iterator) OperatorMinusWithIterator(j iterator) int64 {
	return (int64)(QJsonArray__iterator_OperatorMinusWithIterator(this.h, j))
}

type QJsonArray__const_iterator struct {
	h          uintptr
	isSubclass bool
}

// NewQJsonArray__const_iterator constructs a new QJsonArray::const_iterator object.
func NewQJsonArray__const_iterator() *QJsonArray__const_iterator {

	ret := newQJsonArray__const_iterator(QJsonArray__const_iterator_new())
	ret.isSubclass = true
	return ret
}

// NewQJsonArray__const_iterator2 constructs a new QJsonArray::const_iterator object.
func NewQJsonArray__const_iterator2(array *QJsonArray, index int64) *QJsonArray__const_iterator {

	ret := newQJsonArray__const_iterator(QJsonArray__const_iterator_new2(array.cPointer(), (ptrdiff_t)(index)))
	ret.isSubclass = true
	return ret
}

// NewQJsonArray__const_iterator3 constructs a new QJsonArray::const_iterator object.
func NewQJsonArray__const_iterator3(o *iterator) *QJsonArray__const_iterator {

	ret := newQJsonArray__const_iterator(QJsonArray__const_iterator_new3(o))
	ret.isSubclass = true
	return ret
}

// NewQJsonArray__const_iterator4 constructs a new QJsonArray::const_iterator object.
func NewQJsonArray__const_iterator4(other *const_iterator) *QJsonArray__const_iterator {

	ret := newQJsonArray__const_iterator(QJsonArray__const_iterator_new4(other))
	ret.isSubclass = true
	return ret
}

func (this *QJsonArray__const_iterator) OperatorAssign(other *const_iterator) {
	QJsonArray__const_iterator_OperatorAssign(this.h, other)
}

func (this *QJsonArray__const_iterator) OperatorMultiply() *QJsonValueConstRef {
	_goptr := newQJsonValueConstRef(QJsonArray__const_iterator_OperatorMultiply(this.h))
	_goptr.GoGC() // Qt uses pass-by-value semantics for this type. Mimic with finalizer
	return _goptr
}

func (this *QJsonArray__const_iterator) OperatorMinusGreater() *QJsonValueConstRef {
	return newQJsonValueConstRef(QJsonArray__const_iterator_OperatorMinusGreater(this.h))
}

func (this *QJsonArray__const_iterator) OperatorSubscript(j int64) *QJsonValueConstRef {
	_goptr := newQJsonValueConstRef(QJsonArray__const_iterator_OperatorSubscript(this.h, (ptrdiff_t)(j)))
	_goptr.GoGC() // Qt uses pass-by-value semantics for this type. Mimic with finalizer
	return _goptr
}

func (this *QJsonArray__const_iterator) OperatorPlusPlus() *const_iterator {
	xxxxxxxxx
}

func (this *QJsonArray__const_iterator) OperatorPlusPlusWithInt(param1 int) const_iterator {
	xxxxxxxxx
}

func (this *QJsonArray__const_iterator) OperatorMinusMinus() *const_iterator {
	xxxxxxxxx
}

func (this *QJsonArray__const_iterator) OperatorMinusMinusWithInt(param1 int) const_iterator {
	xxxxxxxxx
}

func (this *QJsonArray__const_iterator) OperatorPlusAssign(j int64) *const_iterator {
	xxxxxxxxx
}

func (this *QJsonArray__const_iterator) OperatorMinusAssign(j int64) *const_iterator {
	xxxxxxxxx
}

func (this *QJsonArray__const_iterator) OperatorPlus(j int64) const_iterator {
	xxxxxxxxx
}

func (this *QJsonArray__const_iterator) OperatorMinus(j int64) const_iterator {
	xxxxxxxxx
}

func (this *QJsonArray__const_iterator) OperatorMinusWithConstIterator(j const_iterator) int64 {
	return (int64)(QJsonArray__const_iterator_OperatorMinusWithConstIterator(this.h, j))
}

File: qt/gen_qjsonarray.cpp
// +build ignore

#include <QJsonArray>
#define WORKAROUND_INNER_CLASS_DEFINITION_QJsonArray__const_iterator
#define WORKAROUND_INNER_CLASS_DEFINITION_QJsonArray__iterator
#include <QJsonValue>
#include <QJsonValueConstRef>
#include <QJsonValueRef>
#include <QList>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <qjsonarray.h>
#include "gen_qjsonarray.h"

#ifndef _Bool
#define _Bool bool
#endif

QJsonArray* QJsonArray_new() {
	return new QJsonArray();
}

QJsonArray* QJsonArray_new2(QJsonArray* other) {
	return new QJsonArray(*other);
}

void QJsonArray_OperatorAssign(QJsonArray* self, QJsonArray* other) {
	self->operator=(*other);
}

QJsonArray* QJsonArray_FromStringList(struct miqt_array /* of struct miqt_string */  list) {
	QStringList list_QList;
	list_QList.reserve(list.len);
	struct miqt_string* list_arr = static_cast<struct miqt_string*>(list.data);
	for(size_t i = 0; i < list.len; ++i) {
		QString list_arr_i_QString = QString::fromUtf8(list_arr[i].data, list_arr[i].len);
		list_QList.push_back(list_arr_i_QString);
	}
	return new QJsonArray(QJsonArray::fromStringList(list_QList));
}

ptrdiff_t QJsonArray_Size(const QJsonArray* self) {
	qsizetype _ret = self->size();
	return static_cast<ptrdiff_t>(_ret);
}

ptrdiff_t QJsonArray_Count(const QJsonArray* self) {
	qsizetype _ret = self->count();
	return static_cast<ptrdiff_t>(_ret);
}

bool QJsonArray_IsEmpty(const QJsonArray* self) {
	return self->isEmpty();
}

QJsonValue* QJsonArray_At(const QJsonArray* self, ptrdiff_t i) {
	return new QJsonValue(self->at((qsizetype)(i)));
}

QJsonValue* QJsonArray_First(const QJsonArray* self) {
	return new QJsonValue(self->first());
}

QJsonValue* QJsonArray_Last(const QJsonArray* self) {
	return new QJsonValue(self->last());
}

void QJsonArray_Prepend(QJsonArray* self, QJsonValue* value) {
	self->prepend(*value);
}

void QJsonArray_Append(QJsonArray* self, QJsonValue* value) {
	self->append(*value);
}

void QJsonArray_RemoveAt(QJsonArray* self, ptrdiff_t i) {
	self->removeAt((qsizetype)(i));
}

QJsonValue* QJsonArray_TakeAt(QJsonArray* self, ptrdiff_t i) {
	return new QJsonValue(self->takeAt((qsizetype)(i)));
}

void QJsonArray_RemoveFirst(QJsonArray* self) {
	self->removeFirst();
}

void QJsonArray_RemoveLast(QJsonArray* self) {
	self->removeLast();
}

void QJsonArray_Insert(QJsonArray* self, ptrdiff_t i, QJsonValue* value) {
	self->insert((qsizetype)(i), *value);
}

void QJsonArray_Replace(QJsonArray* self, ptrdiff_t i, QJsonValue* value) {
	self->replace((qsizetype)(i), *value);
}

bool QJsonArray_Contains(const QJsonArray* self, QJsonValue* element) {
	return self->contains(*element);
}

QJsonValueRef* QJsonArray_OperatorSubscript(QJsonArray* self, ptrdiff_t i) {
	return new QJsonValueRef(self->operator[]((qsizetype)(i)));
}

QJsonValue* QJsonArray_OperatorSubscriptWithQsizetype(const QJsonArray* self, ptrdiff_t i) {
	return new QJsonValue(self->operator[]((qsizetype)(i)));
}

void QJsonArray_Swap(QJsonArray* self, QJsonArray* other) {
	self->swap(*other);
}

iterator QJsonArray_Begin(QJsonArray* self) {
	return self->begin();
}

const_iterator QJsonArray_Begin2(const QJsonArray* self) {
	return self->begin();
}

const_iterator QJsonArray_ConstBegin(const QJsonArray* self) {
	return self->constBegin();
}

const_iterator QJsonArray_Cbegin(const QJsonArray* self) {
	return self->cbegin();
}

iterator QJsonArray_End(QJsonArray* self) {
	return self->end();
}

const_iterator QJsonArray_End2(const QJsonArray* self) {
	return self->end();
}

const_iterator QJsonArray_ConstEnd(const QJsonArray* self) {
	return self->constEnd();
}

const_iterator QJsonArray_Cend(const QJsonArray* self) {
	return self->cend();
}

iterator QJsonArray_Insert2(QJsonArray* self, iterator before, QJsonValue* value) {
	return self->insert(before, *value);
}

iterator QJsonArray_Erase(QJsonArray* self, iterator it) {
	return self->erase(it);
}

QJsonArray* QJsonArray_OperatorPlus(const QJsonArray* self, QJsonValue* v) {
	return new QJsonArray(self->operator+(*v));
}

QJsonArray* QJsonArray_OperatorPlusAssign(QJsonArray* self, QJsonValue* v) {
	QJsonArray& _ret = self->operator+=(*v);
	// Cast returned reference into pointer
	return &_ret;
}

QJsonArray* QJsonArray_OperatorShiftLeft(QJsonArray* self, QJsonValue* v) {
	QJsonArray& _ret = self->operator<<(*v);
	// Cast returned reference into pointer
	return &_ret;
}

void QJsonArray_PushBack(QJsonArray* self, QJsonValue* t) {
	self->push_back(*t);
}

void QJsonArray_PushFront(QJsonArray* self, QJsonValue* t) {
	self->push_front(*t);
}

void QJsonArray_PopFront(QJsonArray* self) {
	self->pop_front();
}

void QJsonArray_PopBack(QJsonArray* self) {
	self->pop_back();
}

bool QJsonArray_Empty(const QJsonArray* self) {
	return self->empty();
}

void QJsonArray_Delete(QJsonArray* self, bool isSubclass) {
	if (isSubclass) {
		delete dynamic_cast<QJsonArray*>( self );
	} else {
		delete self;
	}
}

QJsonArray__iterator* QJsonArray__iterator_new() {
	return new QJsonArray::iterator();
}

QJsonArray__iterator* QJsonArray__iterator_new2(QJsonArray* array, ptrdiff_t index) {
	return new QJsonArray::iterator(array, (qsizetype)(index));
}

QJsonArray__iterator* QJsonArray__iterator_new3(const iterator* other) {
	return new QJsonArray::iterator(*other);
}

void QJsonArray__iterator_OperatorAssign(QJsonArray__iterator* self, const iterator* other) {
	self->operator=(*other);
}

QJsonValueRef* QJsonArray__iterator_OperatorMultiply(const QJsonArray__iterator* self) {
	return new QJsonValueRef(self->operator*());
}

QJsonValueConstRef* QJsonArray__iterator_OperatorMinusGreater(const QJsonArray__iterator* self) {
	return (QJsonValueConstRef*) self->operator->();
}

QJsonValueRef* QJsonArray__iterator_OperatorMinusGreater2(QJsonArray__iterator* self) {
	return self->operator->();
}

QJsonValueRef* QJsonArray__iterator_OperatorSubscript(const QJsonArray__iterator* self, ptrdiff_t j) {
	return new QJsonValueRef(self->operator[]((qsizetype)(j)));
}

iterator* QJsonArray__iterator_OperatorPlusPlus(QJsonArray__iterator* self) {
	return &self->operator++();
}

iterator QJsonArray__iterator_OperatorPlusPlusWithInt(QJsonArray__iterator* self, int param1) {
	return self->operator++(static_cast<int>(param1));
}

iterator* QJsonArray__iterator_OperatorMinusMinus(QJsonArray__iterator* self) {
	return &self->operator--();
}

iterator QJsonArray__iterator_OperatorMinusMinusWithInt(QJsonArray__iterator* self, int param1) {
	return self->operator--(static_cast<int>(param1));
}

iterator* QJsonArray__iterator_OperatorPlusAssign(QJsonArray__iterator* self, ptrdiff_t j) {
	return &self->operator+=((qsizetype)(j));
}

iterator* QJsonArray__iterator_OperatorMinusAssign(QJsonArray__iterator* self, ptrdiff_t j) {
	return &self->operator-=((qsizetype)(j));
}

iterator QJsonArray__iterator_OperatorPlus(const QJsonArray__iterator* self, ptrdiff_t j) {
	return self->operator+((qsizetype)(j));
}

iterator QJsonArray__iterator_OperatorMinus(const QJsonArray__iterator* self, ptrdiff_t j) {
	return self->operator-((qsizetype)(j));
}

ptrdiff_t QJsonArray__iterator_OperatorMinusWithIterator(const QJsonArray__iterator* self, iterator j) {
	qsizetype _ret = self->operator-(j);
	return static_cast<ptrdiff_t>(_ret);
}

void QJsonArray__iterator_Delete(QJsonArray__iterator* self, bool isSubclass) {
	if (isSubclass) {
		delete dynamic_cast<QJsonArray::iterator*>( self );
	} else {
		delete self;
	}
}

QJsonArray__const_iterator* QJsonArray__const_iterator_new() {
	return new QJsonArray::const_iterator();
}

QJsonArray__const_iterator* QJsonArray__const_iterator_new2(QJsonArray* array, ptrdiff_t index) {
	return new QJsonArray::const_iterator(array, (qsizetype)(index));
}

QJsonArray__const_iterator* QJsonArray__const_iterator_new3(const iterator* o) {
	return new QJsonArray::const_iterator(*o);
}

QJsonArray__const_iterator* QJsonArray__const_iterator_new4(const const_iterator* other) {
	return new QJsonArray::const_iterator(*other);
}

void QJsonArray__const_iterator_OperatorAssign(QJsonArray__const_iterator* self, const const_iterator* other) {
	self->operator=(*other);
}

QJsonValueConstRef* QJsonArray__const_iterator_OperatorMultiply(const QJsonArray__const_iterator* self) {
	return new QJsonValueConstRef(self->operator*());
}

QJsonValueConstRef* QJsonArray__const_iterator_OperatorMinusGreater(const QJsonArray__const_iterator* self) {
	return (QJsonValueConstRef*) self->operator->();
}

QJsonValueConstRef* QJsonArray__const_iterator_OperatorSubscript(const QJsonArray__const_iterator* self, ptrdiff_t j) {
	return new QJsonValueConstRef(self->operator[]((qsizetype)(j)));
}

const_iterator* QJsonArray__const_iterator_OperatorPlusPlus(QJsonArray__const_iterator* self) {
	return &self->operator++();
}

const_iterator QJsonArray__const_iterator_OperatorPlusPlusWithInt(QJsonArray__const_iterator* self, int param1) {
	return self->operator++(static_cast<int>(param1));
}

const_iterator* QJsonArray__const_iterator_OperatorMinusMinus(QJsonArray__const_iterator* self) {
	return &self->operator--();
}

const_iterator QJsonArray__const_iterator_OperatorMinusMinusWithInt(QJsonArray__const_iterator* self, int param1) {
	return self->operator--(static_cast<int>(param1));
}

const_iterator* QJsonArray__const_iterator_OperatorPlusAssign(QJsonArray__const_iterator* self, ptrdiff_t j) {
	return &self->operator+=((qsizetype)(j));
}

const_iterator* QJsonArray__const_iterator_OperatorMinusAssign(QJsonArray__const_iterator* self, ptrdiff_t j) {
	return &self->operator-=((qsizetype)(j));
}

const_iterator QJsonArray__const_iterator_OperatorPlus(const QJsonArray__const_iterator* self, ptrdiff_t j) {
	return self->operator+((qsizetype)(j));
}

const_iterator QJsonArray__const_iterator_OperatorMinus(const QJsonArray__const_iterator* self, ptrdiff_t j) {
	return self->operator-((qsizetype)(j));
}

ptrdiff_t QJsonArray__const_iterator_OperatorMinusWithConstIterator(const QJsonArray__const_iterator* self, const_iterator j) {
	qsizetype _ret = self->operator-(j);
	return static_cast<ptrdiff_t>(_ret);
}

void QJsonArray__const_iterator_Delete(QJsonArray__const_iterator* self, bool isSubclass) {
	if (isSubclass) {
		delete dynamic_cast<QJsonArray::const_iterator*>( self );
	} else {
		delete self;
	}
}

